### PR TITLE
[UI smoke test] For #6531: Ads new context menu UI tests.

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Also, needed to upgrade the MockWebServer version.
Added the AndroidAssetDispatcher helper class used on Fenix for serving test pages.
Refactored some robots and test classes.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/ContextMenusTest.kt [hunk 1]
```diff
@@ -36,41 +40,130 @@ class ContextMenusTest {
...
+        fileName = "rabbit.jpg"
+
+        searchScreen {
+        }.loadPage(imagesTestPage.url) {
+            longPressLink(fileName)
+        }.clickSaveImage {
+            // If permission dialog appears, grant it
+            if (permAllowBtn.exists()) {
+                permAllowBtn.click()
+            }
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/ContextMenusTest.kt:153]: I think it should be safe to remove this code on API 30 and use the same runtime grant permissions, worth checking out
``     @Rule
     public GrantPermissionRule permissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
``
https://developer.android.com/reference/android/Manifest.permission.html#READ_EXTERNAL_STORAGE/

**Review by AaronMT — CHANGES_REQUESTED** [review]
Looks ok, see comments first

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt [hunk 2]
```diff
@@ -93,6 +98,21 @@ object TestHelper {
...
+    fun assertAppOpens(packageName: String) {
+        try {
+            if (isPackageInstalled(packageName)) {
+                Intents.intended(IntentMatchers.toPackage(packageName))
+            } else {
+                val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+                mDevice.findObject(UiSelector().textContains("No app found"))
+                    .waitForExists(waitingTime)
+            }
+        } catch (e: ExecutionException) {
```
[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt]: removed this method and used the existing one I didn't notice before

File: app/src/androidTest/java/org/mozilla/focus/activity/ContextMenusTest.kt (hunk 1 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/focus/activity/ContextMenusTest.kt]: done

**Pull request merged**